interface/modules/custom_modules/oe-module-dorn/public/lab_setup.php

```diff
@@ -0,0 +1,209 @@
+                                    <th scope="col"><?php echo xlt("Lab Type") ?></th>
+                                    <th scope="col"><?php echo xlt("Phone Number") ?></th>
+                                    <th scope="col"><?php echo xlt("Fax Number") ?></th>
+                                    <th scope="col"><?php echo xlt("Address") ?></th>
+                                    <th scope="col"><?php echo xlt("City") ?></th>
+                                    <th scope="col"><?php echo xlt("State") ?></th>
+                                    <th scope="col"><?php echo xlt("Zip") ?></th>
+                                    <th scope="col"><?php echo xlt("Compendium Update Date") ?></th>
+                                    <th scope="col"><?php echo xlt("Compendium Download Date") ?></th>
+                                    <th scope="col"><?php echo xlt("Active Routes") ?></th>
+                                    <th scope="col"> <?php echo xlt("Actions") ?> </th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                <?php
+                                foreach ($datas as $data) {
+                                    ?>
+                                    <tr>
+                                        <td scope="row"><?php echo text($data->name); ?></td>
+                                        <td scope="row"><?php echo text($data->labTypeName); ?></td>
+                                        <td scope="row"><?php echo text($data->phoneNumber); ?></td>
+                                        <td scope="row"><?php echo text($data->faxNumber); ?></td>
+                                        <td scope="row"><?php echo text($data->address1); ?> <?php echo text($data->address2); ?> </td>
+                                        <td scope="row"><?php echo text($data->city); ?></td>
+                                        <td scope="row"><?php echo text($data->state); ?></td>
+                                        <td scope="row"><?php echo text($data->zipCode); ?></td>
+                                        <td scope="row"><?php echo text(substr($data->lastCompendiumUpdateDate, 0, 10));?></td>
+                                        <td scope="row"><?php echo text($data->compendiumDownloadDateTime); ?></td>
+                                        <td scope="row"><?php echo text($data->numberOfActiveRoutes); ?></td>
+                                        <td scope="row">
+                                            <button type="button" class="btn btn-primary" onclick="createRouteclick_edit('<?php echo text($data->labGuid); ?>' )"><?php echo xlt('Create Route');?></button>
+                                            <button type="button" class="btn btn-primary" onclick="installCompendiumClick('<?php echo text($data->labGuid); ?>' )"><?php echo xlt('Install Compendium');?></button>
```

Comment:
for escaping stuff in the onclick, use the attr_js() function and remove the quotes from around them:
```
                                            <button type="button" class="btn btn-primary" onclick="createRouteclick_edit(<?php echo attr_js($data->labGuid); ?> )"><?php echo xlt('Create Route');?></button>
                                            <button type="button" class="btn btn-primary" onclick="installCompendiumClick(<?php echo attr_js($data->labGuid); ?> )"><?php echo xlt('Install Compendium');?></button>
```

Reply:
done